hy/completer.py

```diff
@@ -60,8 +60,11 @@
 
 class Completer(object):
     def __init__(self, namespace=None):
-        if namespace and not isinstance(namespace, dict):
```

Comment:
A dictionary has a falsy value when it's empty. So we have to check with `is not None`.